### PR TITLE
Ogrprovider sql logging

### PR DESCRIPTION
**Pull request by elpaso**
GDAL now supports SQLite based layers query logging callback, time for QGIS to use it for debug query logging.

### CODE REVIEW

File: src/core/providers/ogr/qgsogrfeatureiterator.cpp [hunk 1]
```diff
@@ -279,6 +277,23 @@ QgsOgrFeatureIterator::QgsOgrFeatureIterator( QgsOgrFeatureSource *source, bool
     std::sort( mRequestAttributes.begin(), mRequestAttributes.end() );
   }
 
+  // Install query logger
+#if GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION(3,7,0)
+  GDALDatasetSetQueryLoggerFunc( mConn->ds, [ ]( const char *pszSQL, const char *pszError, int64_t lNumRecords, int64_t lExecutionTimeMilliseconds, void *pQueryLoggerArg )
```
Comment (rouault) [inline src/core/providers/ogr/qgsogrfeatureiterator.cpp:282]: for performance reasons, shoudn't this be installed only if SQL logging is enabled ?

Comment (elpaso) [reply src/core/providers/ogr/qgsogrfeatureiterator.cpp]: done

**Comment by rouault** [comment]
If you want to log edits to a dataset, you'd also need to install a logger on the dataset hold by QgsOgrProvider. Perhaps in OpenHelper() in qgsogrproviderutils.cpp, or in QgsOgrProvider::open() by retrieving the dataset handle with ``mOgrLayer->getDatasetHandleAndMutex( datasetMutex )``

**Comment by elpaso** [comment]
> If you want to log edits to a dataset, you'd also need to install a logger on the dataset hold by QgsOgrProvider. Perhaps in OpenHelper() in qgsogrproviderutils.cpp, or in QgsOgrProvider::open() by retrieving the dataset handle with `mOgrLayer->getDatasetHandleAndMutex( datasetMutex )`

There is a problem with this approach: now than the callback is installed only when the query logging is active (which is **not** the usual pattern in QGIS for all other providers) as long as it is installed in the iterator there is not a big problem because the iterator life cycle is kind of short. By installing the callback in the provider (that has a longer life cycle) there is no way for the user to activate the logging if the provider was created when the logging was disabled. 

It's not a big issue of course but this is still different for what all other providers do.

I could install the logger twice, on the provider first and also in the iterator (as it is now) but it seems to me a bit redundant.

I cannot see a way out: installing/disinstalling the callback in the OGR dataset on demand would require additional API in order for the providers to react to a signal fired by the query logger and I'm not sure I will have the time to implement all this new API (without funds).

**Comment by rouault** [comment]
> I cannot see a way out: installing/disinstalling the callback in the OGR dataset on demand would require additional API in order for the providers to react to a signal fired by the query logger and I'm not sure I will have the time to implement all this new API (without funds).

no problem. Just wanting to mention this for completness. This PR looks fine to me as it

**Review by rouault — APPROVED** [review]

**Pull request merged**